projects/politie/ngx-sherlock/src/lib/auto-change-detector.service.ts

```diff
@@ -0,0 +1,101 @@
+import { ChangeDetectorRef, Injectable } from '@angular/core';
+import { _internal } from '@politie/sherlock';
+
+/** @internal */
+export const { symbols } = _internal;
+
+/**
+ * This service wraps the ChangeDetector to automatically detect changes when the state of any used Derivable or DerivableProxy changes.
+ *
+ * Can be used with DI by providing it explicitly in every component that uses it, as follows:
+ *
+ * ```typescript
+ * @Component({
+ *   selector: 'my-component',
+ *   changeDetection: ChangeDetectionStrategy.OnPush,
+ *   providers: [AutoChangeDetectorService], // <-- this is important
+ * })
+ * export class AutoChangeDetectionServiceComponent {
+ *   constructor(acd: AutoChangeDetectorService) {
+ *     acd.init();
+ *   }
+ * }
+ * ```
+ *
+ * Or as follows:
+ * ```typescript
+ * @Component({
+ *   selector: 'my-component',
+ *   changeDetection: ChangeDetectionStrategy.OnPush,
+ * })
+ * export class AutoChangeDetectionServiceComponent {
+ *   constructor(cd: ChangeDetectorRef) {
+ *     new AutoChangeDetectorService(cd).init();
+ *   }
+ * }
+ * ```
+ */
+@Injectable()
+export class AutoChangeDetectorService {
+
+    constructor(private readonly detector: ChangeDetectorRef) {
+        if (!hasOnDestroy(detector)) {
+            // Incompatible Angular version?
+            throw new Error(`Incompatible ChangeDetectorRef`);
```

Comment:
This doesn't feel like a proper error message.

```suggestion
            throw new Error(`The method 'onDestroy' was not found on the provided ChangeDetectorRef. AutoChangeDetectorService will not work correctly.`);
```